ios/iosHackthonApp/Screens/Login/LoginView.swift

```diff
@@ -97,5 +103,27 @@ struct LoginView: View {
         message: Text(Strings.Login.invalidLoginMessage),
         dismissButton: .default(Text(Strings.Common.ok)))
     }
+    .onReceive(self.appState.$moveToHome) { moveToHome in
+      if moveToHome {
+        self.resetOnLogOut()
+      }
+    }
+  }
+}
+
+// MARK: - Functions
+private extension LoginView {
+
+  func resetOnLogOut() {
+    appState.moveToHome = false
+    isSignUpButtonPressed = false
+    loginViewModel.authorizationStore.isAuthorised = false
+    loginViewModel.clear()
+    resetEmailAndPassword()
```

Comment:
This whole `resetOnLogout` it's doing too much for what the name says, IMO.

It's clearing some states, logging out the user, resetting data and also triggering the navigation.

Maybe it's me being completely new to Swift/iOS, but I'd find this a bit confusing to follow.
Something like you did with `resetEmailAndPassword()` where you bundle some small functionality would help. For example: `logoutUserAndNavigateHome()` and `resetUI()` ?

Reply:
yup agreed..I ll try and simplify this

Reply:
broke this to `resetAuthorizationStoreAndLogout` and `resetEmailAndPassword`